src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -170,8 +171,16 @@ internal static bool AddExtraChainCertificates(SafeSslHandle sslContext, X509Cha
             Debug.Assert(chain != null, "X509Chain should not be null");
             Debug.Assert(chain.ChainElements.Count > 0, "chain.Build should have already been called");
 
-            // Don't count the last item (the root)
-            int stop = chain.ChainElements.Count - 1;
+            // If not trusted then include the last item (root). Otherwise, don't include it.
```

Comment:
It's not really trust or not, it's whether it finished with a self-signed certificate.

```suggestion
            // If the last certificate is a root certificate, don't send it. PartialChain means the last cert wasn't a root.
```